cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1997,6 +2000,86 @@ componentOptionalStanza (Cabal.CTestName  _) = Just TestStanzas
 componentOptionalStanza (Cabal.CBenchName _) = Just BenchStanzas
 componentOptionalStanza _                    = Nothing
 
+------------------------------------
+-- Support for --only-dependencies
+--
+
+-- | Try to remove the given targets from the install plan.
+--
+-- This is not always possible.
+--
+pruneInstallPlanToDependencies :: Set UnitId
+                               -> ElaboratedInstallPlan
+                               -> Either CannotPruneDependencies
+                                         ElaboratedInstallPlan
+pruneInstallPlanToDependencies pkgTargets installPlan =
+    assert (all (isJust . InstallPlan.lookup installPlan)
```

Comment:
Yeah. I added it in the integration-tests2 previously, but we should have it in integration-tests and unit-tests